android/src/main/kotlin/io/didomi/fluttersdk/EntitiesHelper.kt

```diff
@@ -0,0 +1,42 @@
+package io.didomi.fluttersdk
+import com.google.gson.Gson
+import io.didomi.sdk.Purpose
+import io.didomi.sdk.Vendor
+import java.util.*
+
+// Class used to convert Didomi entities (Vendor, Purpose) from and to types that can be sent through the Flutter channels.
+class EntitiesHelper {
+    companion object {
+        private val gson = Gson()
+        
+        // Convert a set of purposes into a list of maps.
+        @kotlin.jvm.JvmName("toListOfPurposes")
+        fun toList(purposes: Set<Purpose>?): List<*>? {
+            val valid = purposes ?: return null
```

Comment:
Is the result optional in dart code too? If yes, i agree with this implementation, but if it is not, we should return an empty list instead.

Reply:
currently it's not but I'll update it to be aligned with this. If the set being passed (purposes) is null, I think for consistency we should pass along that value to dart.